misc-scripts/canonical_transcripts/select_canonical_transcripts.pl

```diff
@@ -248,10 +248,36 @@
     my $gene_update_sql = "UPDATE gene SET canonical_transcript_id = ? where gene_id = ?";
     my $gene_sth = $dba->dbc->prepare($gene_update_sql);
 
+    # Prepare transcript canonical attribute sqls
+    my $trans_select_sth = $dba->dbc->prepare("SELECT value FROM transcript_attrib WHERE transcript_id=? AND attrib_type_id=?");
+    my $trans_update_sth = $dba->dbc->prepare("UPDATE transcript_attrib SET value=? WHERE transcript_id=? AND attrib_type_id=?");
+    my $trans_delete_sth = $dba->dbc->prepare("DELETE FROM transcript_attrib WHERE transcript_id=? AND attrib_type_id=?");
+
+    # Get attribute id for canonical
+    my $attrib_type_id_sth = $dba->dbc->prepare("SELECT attrib_type_id FROM attrib_type WHERE code=?");
+    $attrib_type_id_sth->execute('is_canonical');
+    my ($attrib_type_id) = $attrib_type_id_sth->fetchrow_array() || 554;
+
     print "Updating database with new canonical transcripts...\n";
     foreach my $change (@change_list) {
         print "Changin' ". $change->[1]. " on ". $change->[0]."\n" if $verbose;
         $gene_sth->execute( $change->[1], $change->[0]);
+
+        # Check if new canonical transcript attribute exists
+        $trans_select_sth->execute($change->[1], $attrib_type_id);
+        if (my ($new_canonical_exists) = $trans_select_sth->fetchrow_array()) {
```

Comment:
What happens here if the new canonical transcript does not have an `is_canonical` attrib, so no rows are returned by this query?
What happens if there is a row, but the value of `value` is 0?

Reply:
If no rows are returned, the code will not enter into the if statement and thus won't update anything, which I think is the desired outcome. As for the value, in my opinion it doesn't matter what it is, 0 or otherwise. The code will still go into the if statement and update the row even if the value is 0. Our concern here is with its existence and we want to update the value to 1 either way.